views/funcs.html

```diff
@@ -1,36 +1,175 @@
+              <line x1="4.22" y1="19.78" x2="5.64" y2="18.36"></line>
+              <line x1="18.36" y1="5.64" x2="19.78" y2="4.22"></line>
+            </svg>
+            <svg
+              xmlns="http://www.w3.org/2000/svg"
+              width="32"
+              height="32"
+              viewBox="0 0 24 24"
+              fill="none"
+              stroke="currentColor"
+              stroke-width="2"
+              stroke-linecap="round"
+              stroke-linejoin="round"
+              id="theme-toggle-moon"
+            >
+              <path d="M21 12.79A9 9 0 1 1 11.21 3 7 7 0 0 0 21 12.79z"></path>
+            </svg>
+          </button>
+        </div>
+      </div>
+    </div>
+  </nav>
+</div>
+{{ end }} {{ define "html_head" }}
+<meta name="viewport" content="width=device-width,initial-scale=1" />
+<link rel="stylesheet" href="/static/css/normalize.css" />
+<link rel="stylesheet" href="/static/css/app.css" />
+<noscript>
+  <style type="text/css">
+    #source {
+      display: block;
+    }
+    #theme-toggle {
+      display: none;
+    }
+  </style>
+</noscript>
+<script>
+  (function () {
+    const storageKey = "website_theme";
+    const themeDark = "dark";
+    const themeLight = "light";
+
+    function getColorPreference() {
+      if (localStorage.getItem(storageKey)) return localStorage.getItem(storageKey);
+      return window.matchMedia("(prefers-color-scheme: dark)").matches ? themeDark : themeLight;
+    }
+    function setTheme(value) {
+      document.firstElementChild.setAttribute("data-theme", value);
+    }
+    setTheme(getColorPreference());
+    window.matchMedia("(prefers-color-scheme: dark)").addEventListener("change", function (e) {
+      if (!localStorage.getItem(storageKey)) setTheme(e.matches ? themeDark : themeLight);
+    });
+    window.addEventListener("load", function () {
+      document.getElementById("theme-toggle").addEventListener("click", function (e) {
+        e.preventDefault();
+        const newValue = getColorPreference() === themeLight ? themeDark : themeLight;
+        localStorage.setItem(storageKey, newValue);
+        setTheme(newValue);
+      });
+    });
+  })();
+</script>
+{{ end }} {{ define "header_logo" }}
+<a id="logo" href="/">
+  <img src="/static/img/logo.png" alt="Gno.land" title="Gno.land" height="38" />
+</a>
+{{ end }} {{ define "footer" }}
+<footer class="footer">
+  <img src="/static/img/logo.png" alt="Gno.land" width="70" />
+</footer>
+{{ end }} {{ define "js" }}
+<script type="text/javascript" src="/static/js/marked.min.js"></script>
+<script type="text/javascript" src="/static/js/purify.min.js"></script>
+<script type="text/javascript" src="/static/js/renderer.js"></script>
+<script type="text/javascript">
+  function main() {
+    const parsed = parseContent(document.getElementById("source").innerHTML);
+    const DOM = {
+      home: document.getElementById("home"),
+      realm_render: document.getElementById("realm_render"),
```

Comment:
this one is only for chains, not www.

Reply:
You mean the last line only right? We need the 3 first ones. I'm gonna remove the last one.